### PR TITLE
Implement search suggestion test.

### PR DESCRIPTION
**Pull request by josephyu1110**
- home_006

### CODE REVIEW

**Comment by mTwTm** [comment]
Hi Joseph just to let you know we're only looking at bitrise integration now since there are some false-positive in greenhouseci/nevercode. In other words, I think your pr is in good shape in terms of our CI routine.

File: app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static org.mozilla.focus.utils.AndroidTestUtils.getResourceId;
+
+@Keep
+@RunWith(AndroidJUnit4.class)
+public class SearchSuggestionTest {
+
+    @Rule
+    public final ActivityTestRule<MainActivity> mActivityTestRule = new ActivityTestRule<>(MainActivity.class);
+
```
Comment (cnevinc) [inline app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java]: please remove the m prefix

File: app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    @After
+    public void tearDown() {
+        mActivityTestRule.getActivity().finishAndRemoveTask();
+    }
+
+    @Test
+    public void searchSuggestionTest() throws UiObjectNotFoundException, UnsupportedEncodingException {
+
+        SessionLoadedIdlingResource loadingIdlingResource = new SessionLoadedIdlingResource(mActivityTestRule.getActivity());
```
Comment (cnevinc) [inline app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java]: nti: final

File: app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+        // Check if the soft keyboard is shown
+        InputMethodManager inputMethodManager = (InputMethodManager) mContext.getSystemService(Context.INPUT_METHOD_SERVICE);
+        Assert.assertTrue(inputMethodManager.isAcceptingText());
+
+        onView(withId(R.id.url_edit)).perform(typeText("rocket"));
+
+        onView(withId(R.id.search_suggestion)).check(matches(isDisplayed()));
+        // Check if the suggestion count is shown at most 5
+        UiCollection suggestionCollection = new UiCollection(new UiSelector().resourceId(getResourceId("search_suggestion")));
```
Comment (cnevinc) [inline app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java]: nit: final

File: app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        Assert.assertTrue(suggestionCollection.exists());
+        int suggestionCount = suggestionCollection.getChildCount(new UiSelector().resourceId(getResourceId("suggestion_item")));
```
Comment (cnevinc) [inline app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java]: nit: final

File: app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        Assert.assertTrue(suggestionCount <= 5);
+
+        // Pick a suggestion
+        UiObject suggestionItem = suggestionCollection.getChild(new UiSelector().index(new Random().nextInt(suggestionCount)));
```
Comment (cnevinc) [inline app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java]: nit: final ( same for below local variables)

File: app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        SearchEngine defaultSearchEngine = SearchEngineManager.getInstance().getDefaultSearchEngine(mContext);
+
+        // After page loading completes
+        IdlingRegistry.getInstance().register(loadingIdlingResource);
+
+        UiObject urlBar = mUiDevice.findObject(new UiSelector().resourceId(getResourceId("display_url")));
+        Assert.assertTrue(urlBar.exists());
+
+        // Check if the search result is using default search engine
+        Assert.assertTrue(urlBar.getText().contains(defaultSearchEngine.getIdentifier()));
```
Comment (cnevinc) [inline app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java]: Maybe we can find a better way for this check. Cause not every search engine will contain their identifiers in the url. Please add a TODO here and make sure current default search engine is Google in this test case.

**Pull request merged**